recipes/azure-core-cpp/meta.yaml

```
@@ -0,0 +1,65 @@
+{% set version = "1.10.0" %}
+{% set sha256 = "e6ae26ad30376bef6272024564764ba7bf5756063dc94cc4b74180cd004ab75e" %}
+
+package:
+  name: azure-core-cpp
+  version: {{ version }}
+
+source:
+  url: https://github.com/Azure/azure-sdk-for-cpp/archive/refs/tags/azure-core_{{ version }}.tar.gz
+  sha256: {{ sha256 }}
+  patches:
+    # https://github.com/Azure/azure-sdk-for-cpp/issues/4784
+    # https://github.com/Azure/azure-sdk-for-cpp/pull/4785
+    - 0001-remove-wil-from-exported-config.patch
+
+build:
+  number: 0
+  run_exports:
+    - {{ pin_subpackage("azure-core-cpp", min_pin="x", max_pin="x") }}
```

```suggestion
  run_exports:
    - {{ pin_subpackage("azure-core-cpp", max_pin="x") }}
```
Please update all your exports to unspecify the min_pin. SameMajorVersion means backwards compatability, not forward and backward compatability.

---

Also, shouldn't we pin the minor version instead?
```suggestion
  run_exports:
    - {{ pin_subpackage("azure-core-cpp", min_pin="x", max_pin="x.x") }}
```

This suggestion also applies to the other libraries' recipe.

---

We currently have it set to pin to only the major version because the [upstream CMake file](https://github.com/Azure/azure-sdk-for-cpp/blob/2f3e5de931c3f1c19551f6f58418c85aa410745b/cmake-modules/AzureVcpkg.cmake#L152-L156) declares [SameMajorVersion](https://cmake.org/cmake/help/latest/module/CMakePackageConfigHelpers.html) compatibility. See also the previous comments https://github.com/conda-forge/staged-recipes/pull/23297#issuecomment-1629101717 and https://github.com/conda-forge/staged-recipes/pull/23297#issuecomment-1659338135. @jjerphan what do you think of this choice? Pinning to the minor version would be the more conservative route, but it would also require more maintenance when migrating the pins

---

I see. I think we better wait for an actual administrator of conda-forge. If you and another maintainer agreed on pinning on the major version already, just do ignore my comment.